### PR TITLE
Refs [TASK][35435] Handle event News Screen

### PR DESCRIPTION
**Pull request by phucsaker99**
## Related Tickets
- [#Task 35435: Handle event News Screen](https://edu-redmine.sun-asterisk.vn/issues/35435)
## WHAT
- Search event clicked: get api return data list
- BindData : get dataa pour into recyclerview
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/soccert/ui/detailmatch/DetailMatchFragment.kt [hunk 1]
```diff
@@ -56,8 +52,10 @@ class DetailMatchFragment : BaseFragment<FragmentMatchEventBinding>() {
     }
 
     private fun initEventDate() {
-        textFromDate.text = LocalDate.now().minusDays(10).toString()
-        textToDate.text = LocalDate.now().plusDays(10).toString()
+        binding.apply {
+            textFromDate.text = LocalDate.now().minusDays(10).toString()
+            textToDate.text = LocalDate.now().plusDays(10).toString()
```
Comment (Domhoan) [inline app/src/main/java/com/example/soccert/ui/detailmatch/DetailMatchFragment.kt]: hardcode

File: app/src/main/java/com/example/soccert/ui/databinding/ImageBinding.kt [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
-    if (cropStatus) {
-        imageView.loadImageCircle(urlImage)
+fun ImageView.loadUrlImageCircle(urlImage: String?, cropCircle: Boolean = false) {
+    if (cropCircle) {
+        urlImage?.let { loadImageCircle(it) }
     } else {
-        imageView.loadImageRectangle(urlImage)
+        urlImage?.let { loadImageRectangle(it) }
     }
 }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/soccert/ui/databinding/ImageBinding.kt]: check 1 lan ben ngoai thoi 
if(urlImage == null){
    setImageDrawable(null)
    return 
}

File: app/src/main/java/com/example/soccert/ui/home/matchevent/MatchEventFragment.kt [hunk 3]
```diff
@@ -58,8 +52,10 @@ class MatchEventFragment : BaseFragment<FragmentMatchEventBinding>() {
     }
 
     private fun initEventDate() {
-        textFromDate.text = LocalDate.now().minusDays(10).toString()
-        textToDate.text = LocalDate.now().plusDays(10).toString()
+        binding.apply {
+            textFromDate.text = LocalDate.now().minusDays(10).toString()
+            textToDate.text = LocalDate.now().plusDays(10).toString()
```
Comment (Domhoan) [inline app/src/main/java/com/example/soccert/ui/home/matchevent/MatchEventFragment.kt]: hardcode

File: app/src/main/java/com/example/soccert/ui/news/NewsWebFragment.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        binding.apply {
+            lifecycleOwner = viewLifecycleOwner
+        }
+    }
+
+    override fun initActions() {
+        viewModel.error.observe(viewLifecycleOwner, Observer {
+            context?.showToast(ToastType.Error, it)
+        })
+    }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/soccert/ui/news/NewsWebFragment.kt:36]: sao không cho vào base ?

File: app/src/main/res/layout/fragment_news.xml [hunk 5]
```diff
@@ -3,6 +3,13 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools">
 
+    <data>
+
+        <variable
+            name="NewsVM"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_news.xml:9]: sao không đặt tên là viewModel 
```suggestion
            name="viewModel "
```

File: app/src/main/java/com/example/soccert/ui/news/NewsFragment.kt [hunk 6]
```diff
@@ -24,15 +33,47 @@ class NewsFragment : BaseFragment<FragmentNewsBinding>() {
...
     override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         inflater.inflate(R.menu.menu_search, menu)
         super.onCreateOptionsMenu(menu, inflater)
     }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.text_menu_search -> {
+                findNews(item)
+            }
```
Comment (thonc-2135) [inline app/src/main/java/com/example/soccert/ui/news/NewsFragment.kt]:
```suggestion
            R.id.text_menu_search ->  findNews(item)
```
nếu gọn trong 1 hàm r thì viêt liền thế này luôn

File: app/src/main/java/com/example/soccert/ui/detailmatch/DetailMatchFragment.kt (hunk 1 above)
Comment (phucsaker99) [reply app/src/main/java/com/example/soccert/ui/detailmatch/DetailMatchFragment.kt]: cứ để số mình phải định nghĩa hết a?

File: app/src/main/java/com/example/soccert/ui/news/NewsWebFragment.kt (hunk 4 above)
Comment (phucsaker99) [reply app/src/main/java/com/example/soccert/ui/news/NewsWebFragment.kt:36]: vâng a :v e ko nghĩ ra

File: app/src/main/java/com/example/soccert/ui/news/NewsFragment.kt (hunk 6 above)
Comment (phucsaker99) [reply app/src/main/java/com/example/soccert/ui/news/NewsFragment.kt]: vâng a <like

File: app/src/main/res/layout/fragment_news.xml (hunk 5 above)
Comment (phucsaker99) [reply app/src/main/res/layout/fragment_news.xml:9]: e nghĩ viewModel cũng cần phân biệt giữa các fragment khác nhau ?

**Pull request merged**